cmd/tusd/cli/hooks/grpc.go

```diff
@@ -0,0 +1,76 @@
+package hooks
+
+import (
+	"context"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	grpc_retry "github.com/grpc-ecosystem/go-grpc-middleware/retry"
+	"github.com/tus/tusd/pkg/handler"
+	pb "github.com/tus/tusd/pkg/proto/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/status"
+)
+
+type GrpcHook struct {
+	Endpoint   string
+	MaxRetries int
+	Backoff    int
+	GrpcOpts   []grpc.DialOption
+}
+
+func (g GrpcHook) Setup() error {
+	opts := []grpc_retry.CallOption{
+		grpc_retry.WithBackoff(grpc_retry.BackoffLinear(time.Duration(g.Backoff) * time.Second)),
+		grpc_retry.WithMax(uint(g.MaxRetries)),
+	}
+	g.GrpcOpts = []grpc.DialOption{
+		grpc.WithInsecure(),
+		grpc.WithUnaryInterceptor(grpc_retry.UnaryClientInterceptor(opts...)),
+	}
+	return nil
+}
+
+func (g GrpcHook) InvokeHook(typ HookType, info handler.HookEvent, captureOutput bool) ([]byte, int, error) {
+	ctx := context.Background()
+	conn, err := grpc.Dial(g.Endpoint, g.GrpcOpts...)
```

Comment:
Is it necessary to recreate the connection and client for every hook invocation? If not, we can create them once in the Setup function and reuse them here (assuming their are safe to be used by multiple goroutines concurrently).

Reply:
> Is it necessary to recreate the connection and client for every hook invocation?

No, we can reuse the connection. I'll move it to Setup function.

> assuming their are safe to be used by multiple goroutines concurrently

Yes they are